docs/workflow/requirements/linux-requirements.md

```diff
@@ -89,6 +90,6 @@ Install the following packages for the toolchain:
 
 A single line to install all packages above:
 
-    ~$ sudo apt-get install cmake llvm-9 clang-9 autoconf automake libtool build-essential python curl git lldb-6.0 liblldb-6.0-dev libunwind8 libunwind8-dev gettext libicu-dev liblttng-ust-dev libssl-dev libnuma-dev libkrb5-dev
+    ~$ sudo apt-get install cmake llvm-9 clang-9 clang autoconf automake libtool build-essential python curl git lldb-6.0 liblldb-6.0-dev libunwind8 libunwind8-dev gettext libicu-dev liblttng-ust-dev libssl-dev libnuma-dev libkrb5-dev
```

Comment:
This is strange. @marek-safar cannot mono be built with clang-9 like coreclr? It would be unfortunate to require two different clang versions.

Reply:
@davidwrighton could you share the build error you hit?

Reply:
This is the error. It happens while running autoconf. If I install the clang package the problem goes away, and mono appears to build correctly. Presumably there is something that should cause mono to be configured to build with clang-9, but that isn't what I see happening.
```
davidwr@davidwr-h-001:~/runtime$ ./build.sh --subsetCategory mono -c Release
Downloading 'https://dot.net/v1/dotnet-install.sh'
dotnet_install: Warning: Unable to locate zlib. Probable prerequisite missing; install zlib.
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-alpha.1.20073.4/dotnet-sdk-5.0.100-alpha.1.20073.4-linux-x64.tar.gz
dotnet-install: Extracting zip from https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-alpha.1.20073.4/dotnet-sdk-5.0.100-alpha.1.20073.4-linux-x64.tar.gz
dotnet-install: Adding to current process PATH: `/home/davidwr/runtime/.dotnet`. Note: This change will be visible only when sourcing script.
dotnet-install: Installation finished successfully.
  Restore completed in 1.23 sec for /home/davidwr/runtime/tools-local/tasks/installer.tasks/installer.tasks.csproj.
  installer.tasks -> /home/davidwr/runtime/artifacts/bin/installer.tasks/Debug/netstandard2.0/installer.tasks.dll
  Tool 'coverlet.console' (version '1.7.0') was restored. Available commands: coverlet
  Tool 'dotnet-reportgenerator-globaltool' (version '4.4.2') was restored. Available commands: reportgenerator
  Tools restore was successful.
  Restore completed in 54.32 ms for /home/davidwr/.nuget/packages/microsoft.dotnet.arcade.sdk/5.0.0-beta.20079.8/tools/Tools.proj.
  Restore completed in 42.03 ms for /home/davidwr/runtime/src/mono/netcore/System.Private.CoreLib/System.Private.CoreLib.csproj.
  Restore completed in 8.04 ms for /home/davidwr/runtime/src/mono/netcore/nuget/packages.builds.
  Restore completed in 9.25 ms for /home/davidwr/runtime/src/mono/netcore/nuget/Microsoft.NETCore.Runtime.Mono/Microsoft.NETCore.Runtime.Mono.builds.
  Configuring Mono with ' --with-core=only CC=clang CXX=clang++ CFLAGS="-O2 -g" CXXFLAGS="-O2 -g"'
  Running libtoolize...
  libtoolize: putting auxiliary files in '.'.
  libtoolize: copying file './ltmain.sh'
  libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
  libtoolize: copying file 'm4/libtool.m4'
  libtoolize: copying file 'm4/ltoptions.m4'
  libtoolize: copying file 'm4/ltsugar.m4'
  libtoolize: copying file 'm4/ltversion.m4'
  libtoolize: copying file 'm4/lt~obsolete.m4'
  Running aclocal -I m4 -I .  ...
  Running autoheader...
  Running automake --add-missing --copy --gnu -Wno-portability -Wno-obsolete ...
  configure.ac:609: installing './compile'
  configure.ac:13: installing './config.guess'
  configure.ac:13: installing './config.sub'
  configure.ac:20: installing './install-sh'
  configure.ac:20: installing './missing'
  mono/arch/riscv/Makefile.am: installing './depcomp'
  Running autoconf ...
  Skipping configure process.
  checking build system type... x86_64-pc-linux-gnu
  checking host system type... x86_64-pc-linux-gnu
  checking target system type... x86_64-pc-linux-gnu
  checking for a BSD-compatible install... /usr/bin/install -c
  checking whether build environment is sane... yes
  checking for a thread-safe mkdir -p... /bin/mkdir -p
  checking for gawk... no
  checking for mawk... mawk
  checking whether make sets $(MAKE)... yes
  checking whether make supports nested variables... yes
  checking whether UID '1000' is supported by ustar format... yes
  checking whether GID '1000' is supported by ustar format... yes
  checking how to create a ustar tar archive... gnutar
  checking whether to enable maintainer-specific portions of Makefiles... no
  checking whether ln -s works... yes
  checking whether make supports nested variables... (cached) yes
  checking host platform characteristics... ok
  checking for gcc... clang
  checking for gcc... (cached) clang
  checking whether the C compiler works... no
configure : error : in `/home/davidwr/runtime/artifacts/obj/mono/Linux.x64.Release': [/home/davidwr/runtime/src/mono/mono.proj]
configure : error : C compiler cannot create executables [/home/davidwr/runtime/src/mono/mono.proj]
  See `config.log' for more details
/home/davidwr/runtime/src/mono/mono.proj(40,5): error MSB3073: The command "/home/davidwr/runtime/src/mono/configure  --with-core=only CC=clang CXX=clang++ CFLAGS="-O2 -g" CXXFLAGS="-O2 -g"" exited with code 77.
  System.Private.CoreLib -> /home/davidwr/runtime/artifacts/bin/mono/Linux.x64.Release/IL/System.Private.CoreLib.dll

Build FAILED.

configure : error : in `/home/davidwr/runtime/artifacts/obj/mono/Linux.x64.Release': [/home/davidwr/runtime/src/mono/mono.proj]
configure : error : C compiler cannot create executables [/home/davidwr/runtime/src/mono/mono.proj]
/home/davidwr/runtime/src/mono/mono.proj(40,5): error MSB3073: The command "/home/davidwr/runtime/src/mono/configure  --with-core=only CC=clang CXX=clang++ CFLAGS="-O2 -g" CXXFLAGS="-O2 -g"" exited with code 77.
    0 Warning(s)
    3 Error(s)
```

Reply:
@davidwrighton can you please share what is the Ubuntu version you are using and how you have installed the clang 9?

Reply:
I have "Ubuntu 18.04.3 LTS" installed, and I believe I installed clang-9 by following the instruction to run a sudo apt-get command with whatever parameters were suggested 3-4 months ago. I don't remember in particular.

Reply:
When I saw a failure, I re-ran apt-get with the exact command line as suggested, and it didn't fix the problem, until I looked a little harder and saw that the error message was mentioning `clang`, and not `clang-9`.

Reply:
Are you sure you have clang 9 actually installed? I am asking since it is not available in the default Ubuntu repo for Ubuntu 18.04.3, you need to use LLVM repo to install it. 
I can see now that the doc is missing the details on adding LLVM repo to the repos apt-get can see. We used to have it there, so apparently it was accidentally dropped at some point.

Reply:
@davidwrighton could you please also include your `config.log` content

Reply:
/cc @directhex

Reply:
@davidwrighton you can verify whether clang-9 is installed or not by running "clang-9" command.

Reply:
I do have the LLVM repo set up, and clang-9 is installed.

```
davidwr@davidwr-h-001:~/runtime$ clang-9
clang: error: no input files
```

The config log looks like
https://gist.github.com/davidwrighton/318164edca25175014e13ed714c15bbb

My guess is that something is causing the command line arguments to configure for mono to specify that CC=clang, instead of what should probably be happening which is CC=clang-9, but I dunno.

Reply:
We do it here https://github.com/dotnet/runtime/blob/master/src/mono/mono.proj#L34 which could be changed to clang-9 for Linux only. Where do you set clang version in coreclr?

/cc @akoeplinger

Reply:
Yeah we shouldn't need this. There's logic here which detects the clang version that is available: https://github.com/dotnet/runtime/blob/1bbd51d56a8e4e2895a9b707cd0551ee0b28e253/eng/native/gen-buildsys.sh#L51-L61

Reply:
https://github.com/dotnet/runtime/pull/31854